src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -141,7 +142,8 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
         assert dd.hostname is not None
         addr = dd.ip if dd.ip else self._inventory_get_fqdn(dd.hostname)
         port = dd.ports[0] if dd.ports else self.DEFAULT_SERVICE_PORT
-        service_url = build_url(scheme='https', host=addr, port=port)
+        spec = cast(GrafanaSpec, self.mgr.spec_store[dd.service_name()].spec)
+        service_url = build_url(scheme=spec.protocol, host=addr, port=port)
```

Comment:
> when testing this, the protocol on the URL correctly came out as http, but I got a strange URL nonetheless. I don't think this would be something caused by this PR but have you seen this before?
> 
> ```
> [ceph: root@vm-00 /]# ceph dashboard get-grafana-api-url
> http://host.containers.internal:3000
> ```

@adk3798 not sure where this host comes from. In my testing I've never seen a similar url. Normally I get the hostname of the node where grafana is running.